src/reducers/versions.spec.tsx

```diff
@@ -3135,4 +3240,33 @@ describe(__filename, () => {
       expect(selectCurrentVersionInfo(state)).toEqual(false);
     });
   });
+
+  describe('getPrunedExpandedPaths', () => {
+    it('perserves the paths that exist in the new version', () => {
+      const path1 = 'path/1';
+      const path2 = 'path/2';
+      const expandedPaths = [path1];
+      const version = createExternalVersionWithEntries([
+        createFakeEntry('directory', path1),
+        createFakeEntry('directory', path2),
+      ]);
+
+      expect(
+        getPrunedExpandedPaths(expandedPaths, createInternalVersion(version)),
+      ).toEqual([path1]);
```

Comment:
I like how you made separate tests for inclusion / exclusion. As it is, this assertion covers both inclusion and exclusion so I suggest making this one just about inclusion. Example `expect(...).arrayContaining([path1])`